Updated Help text. Including "usage" in the argparse.ArgumentParser()

DerekFurstPitt opened the pull request:
Updated help text. Originally I was manually setting "usage" in argparse.ArgumentParser(). Apparently this actually overwrote some default behavior for the help text. Removing it makes it behave predictably now. The usage is auto generated, and then if I include one of the subcommands followed by --help it will output arguments and options specific to that sub command. 

This pr also includes a change made to omit the first line of a manifest file if it starts with dataset_id.

shirey reviewed (commented):
Can you change the help text to match the attached?  It is fine to just print it out instead of using the parser's help function.
[clt-help.txt](https://github.com/hubmapconsortium/hubmap-clt/files/8117353/clt-help.txt)

DerekFurstPitt commented:
Made numerous changes. This pr is still up. so I'm re-requesting review. Changes include adding logout feature, changing the default download location, reworked manifest parsing to use a regex, reworked the entire help-text.
